Description: Don't increment stats of foreign device
 The receive hook for batman-adv ethernet frames tried to get the last device
 which processed the skb before us. It only used that information to update the
 rx_bytes and rx_packets stat of that foreign device which already has updated
 it using its own receive functions.
Author: Sven Eckelmann <sven.eckelmann@gmx.de>
Bug-Debian: http://bugs.debian.org/593473

---
diff --git a/hard-interface.c b/hard-interface.c
index bc2bd12cf96ce854311b28fdc3d73e5ecb41edc5..b2c53b771e337b979f7cf7d0054a37778e176a43 100644
--- a/hard-interface.c
+++ b/hard-interface.c
@@ -440,7 +440,6 @@ int batman_skb_recv(struct sk_buff *skb, struct net_device *dev,
 {
 	struct batman_packet *batman_packet;
 	struct batman_if *batman_if;
-	struct net_device_stats *stats;
 	int ret;
 
 	skb = skb_share_check(skb, GFP_ATOMIC);
@@ -469,12 +468,6 @@ int batman_skb_recv(struct sk_buff *skb, struct net_device *dev,
 	if (batman_if->if_status != IF_ACTIVE)
 		goto err_free;
 
-	stats = (struct net_device_stats *)dev_get_stats(skb->dev);
-	if (stats) {
-		stats->rx_packets++;
-		stats->rx_bytes += skb->len;
-	}
-
 	batman_packet = (struct batman_packet *)skb->data;
 
 	if (batman_packet->version != COMPAT_VERSION) {
